pkg/util/util.go

```diff
@@ -45,6 +47,71 @@ func GenerateRandomString(n int) string {
 	return string(b)
 }
 
+// StringInSlice returns if a passed string exists in the passed slice of strings
+func StringInSlice(checkStr string, strSlice []string) bool {
+	// Iterate the slice
+	for _, ele := range strSlice {
+		// if current element is same as element to search(checkStr), return true indicating checkStr found
+		if ele == checkStr {
+			return true
+		}
+	}
+	// Exited loop without finding required element(checkStr) in slice, so return false indicating checkStr not found in slice
+	return false
+}
+
+// GetRandomName returns a randomly generated name which can be used for naming odo and/or openshift entities
+// prefix: Desired prefix part of the name
+// existList: List to verify that the returned name does not already exist
+// preferredSuffix: Optional suffix if passed, will be checked if prefix-suffix does not exist in existList if not, will add an additional timestamp to make name unique
+// Returns randomname is prefix-suffix, if suffix is passed else generated. Aditionally, if the prefix-suffix is used, it'll be appended with additional 4 char string to take the form prefix-suffix-{a-z A-z}4+
+func GetRandomName(prefix string, existList []string, preferredSuffix string) string {
+	if preferredSuffix == "" {
+		// Generate suffix if not passed, using random country names generated from Pallinder/go-randomdata as suffix
+		preferredSuffix = strings.Replace(
```

Comment:
We can use strings.NewReplacer() for this task https://golang.org/pkg/strings/#Replacer

Reply:
Nice one :)
Will use it